packages/filesystem/src/browser/file-resource.ts

```diff
@@ -326,13 +330,21 @@ export class FileResourceResolver implements ResourceResolver {
     }
 
     protected async shouldOpenAsText(uri: URI, error: string): Promise<boolean> {
-        const dialog = new ConfirmDialog({
-            title: error,
-            msg: `Opening it might take some time and might make the IDE unresponsive. Do you want to open '${this.labelProvider.getLongName(uri)}' anyway?`,
-            ok: 'Yes',
-            cancel: 'No'
-        });
-        return !!await dialog.open();
+        switch (this.applicationState.state) {
+            case 'init':
+            case 'started_contributions':
+            case 'attached_shell':
+                return true; // We're restoring state - assume that we should open files that were previously open.
+            default: {
+                const dialog = new ConfirmDialog({
+                    title: error,
+                    msg: `Opening it might take some time and might make the IDE unresponsive. Do you want to open '${this.labelProvider.getLongName(uri)}' anyway?`,
```

Comment:
@msujew this makes me wonder if we are somehow missing some localizations, perhaps we need to enhance the custom eslint rule. 

The same is likely true here (unless it is done automatically that I am not aware of):

https://github.com/eclipse-theia/theia/blob/432602946323bc1500c585de4432021b6d6a4a0a/packages/filesystem/src/browser/file-resource.ts#L320-L321

How should we go about tracking what might be missing?

Reply:
Yeah, I noticed that as well. I mostly looked for widgets, commands, menus etc. but missed a lot of dialogs. I'm still unsure how we write such a eslint rule, without generating tons of false-positives. Only for strings inside of certain dialog constructors (`new ConfirmDialog(...)` etc)?

Reply:
That's true, it might produce too many false-positives, even if we were to only apply the rule to `browser/`. It might be worthwhile however to have a way to run the check manually periodically so we can see if we missed anything, but we would need to also manually verify if they are valid.